### ISSUE TITLE
SearchDestinyPlayer returning incorrect membershipId

### ISSUE THREAD

**Issue by BlazeBoss17** [Bungie-net/api#519]
The call https://www.bungie.net/Platform/Destiny2/SearchDestinyPlayer/-1/informer/ returns the membershipId 4611686018465893351 for playstation, which is a nonexistent membershipId. I believe it should return 4611686018428560404.

**Comment by rentalhost** [comment]
Try to replace `-1` with `4`.

**Comment by floatingatoll** [comment]
Thorn, is the -1 All case actually supported on this endpoint?

http://bungienetplatform.wikia.com/wiki/BungieMembershipType

**Comment by vthornheart-bng** [comment]
Hmm, unfortunately the situation you're running into appears to be due to a user who changed their platform name, and then another user who changed their name to that platform name.  I'll bring this up to some folks internally.

**Comment by vthornheart-bng** [comment]
Hmm, indeed you're correct!  I know we've run into name change-related shenanigans with other platforms, but indeed it's not possible with PSN.  I'm speaking with some folks internally and will loop back if/when I get more information.  But unfortunately I don't have a good workaround for you right now for this user.

**Comment by philliesgamer** [comment]
I wanted to add additional info to this issue since I am the user affected by this issue. My bungie.net account was first created on the Xbox platform when I played Halo. I believe Halo 2 was the earliest Halo game that had bungie.net records.

I started playing Destiny and Destiny 2 on Playstation 4. Both my Xbox gamertag and PSN username were previously linked to my bungie.net account. I have never had reporting issues with Destiny 1, but this has always been an issue with Destiny 2. No 3rd party websites are able to track my username although I can see my characters and stats on the bungie.net website. I unlinked my Xbox gamertag several months ago from my bungie.net account in hopes of fixing this issue, but it did not.

I have never changed my Xbox gamertag or PSN username and they are also the exact same name on both gaming networks (informer).

**Comment by vthornheart-bng** [comment]
Thank you!  I will pass this additional information on!

**Comment by philliesgamer** [comment]
Has there been any updates on this issue? The incorrect call against my playstation membershipId is preventing me from participating in many different activities as I cannot use any 3rd party websites to use or post on LFG websites, run raid reports, check external stats or do other things. It has really impacted my player experience and I would greatly appreciate any resolution to this issue.

**Comment by vthornheart-bng** [comment]
Good question - I've got no actionable update yet unfortunately.  I'm hitting up the game-side folks again to check on the status and see if there's any update or if they have a way of shedding any more light on the situation.

Have you ever had a "forced name change" on either of your accounts (XBox and/or PSN)?  If so, do you know what the "old" name was and what your "new" name is on the platform(s) where it occurred?

Also, when was the last time you logged into Destiny 2?  This could be helpful information as well.

**Comment by vthornheart-bng** [comment]
Also, are you able to use the "official" Companion app successfully?  If you are, could you take a screenshot of your More -> My Profile screen, and tell me what platforms appear in the drop-down box on that screen?

**Comment by philliesgamer** [comment]
I've never had a forced name change on either of my accounts - I actually haven't used my informer account on Xbox One lately, but I play as informer on Playstation 4 daily (and this is where I played Destiny 1 and currently play Destiny 2).

I was a beta tester for Xbox Live so I actually had the informer name before the inception of Xbox Live. For playstation, I chose informer a few weeks after Playstation 3 launch when the Playstation Network was brand new and have never changed any name on there.

The last time I logged into Destiny 2 was probably around 4pm EDT (about 4 hours ago or so).

I am able to use the Destiny 2 official app successfully, but I have to authenticate often. For example, right now when I start it, I see the error message of "No valid Destiny 2 characters were found for your linked platform account. Please try again. ! Tap to retry. Alternatively, you can select another platform - informer on PSN"

I am able to log into the app find after I tap to retry. I have 3 characters at 600 light - hunter, titan and warlock.

I see no dropdown option when I go to More -> My Profile in the Destiny app. I actually unlinked my Xbox account from my bungie.net profile about 4 months ago in trying to troubleshoot this issue and never linked it back (because I don't play Halo games anymore). The only profile I see is my informer PlayStation account.
![getattachmentthumbnail](https://user-images.githubusercontent.com/38703427/47688058-dbed6100-dbb9-11e8-9d1a-59093f0bcc14.png)

**Comment by philliesgamer** [comment]
![getattachmentthumbnail](https://user-images.githubusercontent.com/38703427/47688089-0f2ff000-dbba-11e8-926a-c6d798e8024a.png)

**Comment by floatingatoll** [comment]
My app screenshot has a numeric ID above the word PlayStation, differing from the one above.

![img_0322](https://user-images.githubusercontent.com/92684/47688319-e1d74800-dba1-11e8-835e-2932dee92b90.jpg)

**Comment by floatingatoll** [comment]
The matching profile page for my screenshot at https://www.bungie.net/en/Profile/254/8922309/floatingatoll shows that value as `<div class="unique-name">` which maps to my JSON `uniqueName` value.

```json
GET /platform/User/GetBungieNetUser/?lc=en

{
	"Response": {
		"user": {
			"membershipId": "8922309",
			"uniqueName": "8922309",
			"displayName": "floatingatoll",
```

**Comment by philliesgamer** [comment]
This is my profile page here: [https://www.bungie.net/en/Profile/254/520229/informer](https://www.bungie.net/en/Profile/254/520229/informer)

**Comment by vthornheart-bng** [comment]
Beautiful, that's great info - thank you!

I'll pass this on.  Sorry that this has been such a persistent issue!

**Comment by vthornheart-bng** [comment]
Hey, one more favor I'd like to ask of you.  Could you give a screenshot or picture of your orbit screen in Destiny, with your nameplate showing?

**Comment by philliesgamer** [comment]
I took a picture with my phone, I hope this is sufficient enough. If not, please let me know what else you need. I really appreciate the help!

![thumbnail_image1](https://user-images.githubusercontent.com/38703427/47742964-3509d300-dc54-11e8-9f62-c8c6517d22f7.jpg)

**Comment by vthornheart-bng** [comment]
Yes, that's perfect - thank you!  I wanted to give them 100% evidence that your screenname on the platform was still identical to what Bungie.net was reporting it to be.

**Comment by philliesgamer** [comment]
Any new updates on this? I am still an avid fan and player but this bug is preventing me from using any LFG or raid reporting sites. Any help would be greatly appreciated. Thanks!
